hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/security/SaslPropertiesResolver.java

```diff
@@ -60,16 +62,17 @@ public static SaslPropertiesResolver getInstance(Configuration conf) {
   @Override
   public void setConf(Configuration conf) {
     this.conf = conf;
-    properties = new TreeMap<String,String>();
+    ImmutableMap.Builder<String, String> propertiesBuilder = ImmutableMap.builder();
```

Comment:
Is it important to preserve the TreeMap ordering semantics?

Reply:
Yea I think we should try to keep it as unchanged as possible. I'm not sure why it needs to be a TreeMap, maybe in some contexts the ordering matters? Maybe we should just wrap it in ImmutableMap.copyOf?